src/core/collision.cpp

```diff
@@ -507,6 +513,19 @@ static void three_particle_binding_do_search(Cell *basecell, Particle &p1,
   }
 }
 
+Cell *responsible_collision_cell(Particle& p)
```

Comment:
This should go into `cells.cpp`, and can be reused there e.g. to make removing a particle O(1). It should also be renamed to something like `find_current_cell`. This finds the actual cell a particles is in, as opposed to the cell a particle should be in (which is what `position_to_cell` returns).